symengine/ruby/spec/basic_spec.rb

```diff
@@ -1,71 +1,171 @@
+            expect(c).to be true
+          end
+        end
+      end
+      describe '#eql?' do
+        context 'with another initialised Basic object as argument' do
+          it 'returns true if they are the same expression
+                        false if not' do
+            a = SymEngine::Symbol.new('x')
+            b = SymEngine::Symbol.new('y')
+            c = ((a * b).eql?(@a * @b))
+            expect(c).to be true
+          end
+        end
+      end
+      describe '#!=' do
+        context 'with another initialised Basic object as argument' do
+          it 'returns true is they are not the same expression
+                        false if they are' do |_variable|
+            a = SymEngine::Symbol.new('x')
+            b = SymEngine::Symbol.new('y')
+            c = ((a * b) != (@a * @b))
+            expect(c).to be false
+          end
         end
+      end
+    end
+
+    describe 'unary operations' do
+      before :each do
+        @x = SymEngine::Symbol.new('a')
+      end
+      describe '#-@' do
+        context "doesn't take any argument" do
+          it 'returns the negation of self' do
+            p = -@x
+            expect(p).to be_an_instance_of SymEngine::Symbol
+            expect(p.to_s).to eql('-a')
+          end
+        end
+      end
+    end
+
+    describe '#args' do
+      context 'with nothing as argument' do
+        it 'returns array of arguments' do
+          x = SymEngine::Symbol.new('x')
+          y = SymEngine::Symbol.new('y')
+          z = SymEngine::Symbol.new('z')
+          e = (x**y + z)
+          f = e.args
+          expect(f).to be_an_instance_of Array
+          expect(f.length).to be 2
+          expect(f.to_set).to eql([x**y, z].to_set)
+        end
+      end
+    end
+
+    describe '#free_symbols' do
+      context 'with nothing as argument' do
+        it 'returns the set of symbols' do
+          x = SymEngine::Symbol.new('x')
+          y = SymEngine::Symbol.new('y')
+          z = SymEngine::Symbol.new('z')
+          e = (x**y / z)
+          f = e.free_symbols
+          expect(f).to be_an_instance_of Set
+          expect(f.length).to be 3
+          expect(f).to eql([x, y, z].to_set)
+        end
+      end
+    end
 
-        describe 'unary operations' do
-            describe '#-@' do
-                context "doesn't take any argument" do
-                    xit 'returns the negation of self' do
-                    end
-                end
-            end
+    describe '#expand' do
+      context 'with nothing as argument' do
+        it 'returns the expanded form' do
+          x = SymEngine::Symbol.new('x')
+          y = SymEngine::Symbol.new('y')
+          z = SymEngine::Symbol.new('z')
+          e = (x + y + z) * (x + y + z)
+          f = e.expand
+          expect(e.to_s).to eql('(x + y + z)**2')
+          expect(f).to be_an_instance_of SymEngine::Symbol
+          expect(f.to_s).to eql('2*x*y + 2*x*z + 2*y*z + x**2 + y**2 + z**2')
```

Comment:
Same here, the integers is the only thing that is stopping this from happening. I am adding this as soon as I can.